### PR TITLE
style: add markdown lint

### PR DESCRIPTION
**Pull request by yifanwu-kolena**
### Linked issue(s):

follow up of https://github.com/kolenaIO/kolena/pull/453#discussion_r1447579988

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: docs/metrics/geometry-matching.md [hunk 1]
```diff
@@ -195,17 +192,16 @@ the IoU threshold to be considered as a valid match.
 
 ??? info "Pseudocode: PASCAL VOC Matching"
 
-	1. Loop through all images in your dataset;
```
Comment (yifanwu-kolena) [inline docs/metrics/geometry-matching.md:198]: tabs to spaces

File: .markdownlint.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+default: true
+
+MD010:
```
Comment (ly-kolena) [inline .markdownlint.yaml:5]: Nit: can we use comment to inform what each rule is, so that I don't have to google the library for its meaning?

File: docs/index.md [hunk 3]
```diff
@@ -9,12 +9,14 @@ hide:
   ![Kolena](assets/images/wordmark-white.svg#only-dark){ width="400" }
 </figure>
 
+<!-- markdownlint-disable MD013 -->
```
Comment (gordonhart) [inline docs/index.md]: It's not great to have to disable this all over the place — do we ever need `MD013`?

Comment (yifanwu-kolena) [reply docs/index.md]: [MD013](https://github.com/DavidAnson/markdownlint/blob/main/doc/md013.md) is the rule for line length

File: .markdownlint.yaml (hunk 2 above)
Comment (yifanwu-kolena) [reply .markdownlint.yaml:5]: good idea, just added the comment with the rules url as well

File: .markdownlint.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+default: true
+
+# Hard tabs
+MD010:
+  spaces_per_tab: 4
+
+# Line length
+MD013:
+  code_blocks: false
+  headings: false
```
Comment (ly-kolena) [inline .markdownlint.yaml]: Nit: do we not want to apply this rule to code_blocks and headings?

File: README.md [hunk 5]
```diff
@@ -4,12 +4,14 @@
   </a>
 </p>
 
+<!-- markdownlint-disable MD013 -->
```
Comment (ly-kolena) [inline README.md]: Can this section be line-broken to comply to the length rule without exception? (assuming this can help

> This rule has an exception when there is no whitespace beyond the configured line length. This allows you to include items such as long URLs without being forced to break them in the middle.

**Review by ly-kolena — APPROVED** [review]

File: docs/metrics/wer-cer-mer.md [hunk 6]
```diff
@@ -66,14 +70,20 @@ $$
 $$
 
 ### Example
+
 Let's calculate the word error rate between the following reference and candidate texts:
 
+<!-- markdownlint-disable MD013 -->
```
Comment (ly-kolena) [inline docs/metrics/wer-cer-mer.md]: For my knowledge, why is this needed when we have `table` disabled in MD013 configuration?

Comment (yifanwu-kolena) [reply docs/metrics/wer-cer-mer.md]: after updating the table syntax to have explicit header separator, it is recognized by the linter properly.
in term of other tables nested in the ??? block, it will be recognized as code rather than table

File: .markdownlint.yaml (hunk 4 above)
Comment (yifanwu-kolena) [reply .markdownlint.yaml]: just enabled
